### DISCUSSION TITLE
Lighten CompositeOperator in Magick++

### DISCUSSION THREAD

**Discussion by fahimmehraj** [ImageMagick/ImageMagick#3751] [Help] [answered]
I've been looking to achieve a certain functionality with the Magick++ library, and noticed what I consider to be a gap in the functionality of the library. Using compose from the terminal, you are able to blend two images together using a wide range of modes. In Magick++, only the simplest of these modes are implemented as CompositeOperators. Will the rest of the modes be added at one point? Perhaps I am mistaking the goal of Magick++, are these blend modes not meant to be implemented and left as something to implement for the developer? If it is, I would really appreciate if someone would be willing to point me in the right direction to implement the Lighten composite blend mode.

(for more context on what I'm talking about, see https://stackoverflow.com/questions/43003697/blend-multiple-images-with-imagemagic)

**Comment by snibgo** [comment, marked as answer]
What version of IM?

See MagickCore\composite.h:
```
typedef enum
{
  UndefinedCompositeOp,
  AlphaCompositeOp,
  AtopCompositeOp,
  BlendCompositeOp,
  :
  LightenCompositeOp,
  LightenIntensityCompositeOp,
  :
  XorCompositeOp,
  StereoCompositeOp
} CompositeOperator;
```
Are any operators missing?

**Reply by fahimmehraj** [reply]
oops, I didn't try experimenting in my editor. I just looked at the magick++ documentation and thought that was all

https://imagemagick.org/Magick++/Enumerations.html#CompositeOperator
